### PR TITLE
Introduce Cypress integration tests

### PR DESCRIPTION
**Pull request by gdemonet**
**Component**: ui, tests

**Context**: #2878 

**Summary**:

- Re-organize Cypress tests to isolate E2E tests
- Setup minimal mocks to run Cypress integration tests without a proper back-end
- Add complete test suite for the Navbar component
- Run Cypress integration tests in the CI

**Acceptance criteria**: 

- CI passes

---

See: #2878

### CODE REVIEW

File: ui/cypress/support/commands.js [hunk 1]
```diff
@@ -51,18 +51,18 @@ Cypress.Commands.add('setupMocks', () => {
 const ADMIN_JWT = {
   id_token:
     'eyJhbGciOiJSUzI1NiIsImtpZCI6IjY4MGZmMWVhZTBlZGYyMDI0MDQyOGVjMzRlMjBlODNjNzBlODAwODIifQ.eyJpc3MiOiJodHRwczovLzEwLjIwMC4yLjE1MDo4NDQzL29pZGMiLCJzdWIiOiJDaVF3T0dFNE5qZzBZaTFrWWpnNExUUmlOek10T1RCaE9TMHpZMlF4TmpZeFpqVTBOallTQld4dlkyRnMiLCJhdWQiOlsib2lkYy1hdXRoLWNsaWVudCIsIm1ldGFsazhzLXVpIl0sImV4cCI6MTYwMzQwMDg1MCwiaWF0IjoxNjAzMzE0NDUwLCJhenAiOiJtZXRhbGs4cy11aSIsIm5vbmNlIjoiZTM2MzRkMGQyYmRkNGMwMzgyNzllNmIzOWMxYjI0YzYiLCJhdF9oYXNoIjoiV3ZBSlFLQ3BKVVAwMHFjdUpCQWpCUSIsImVtYWlsIjoiYWRtaW5AbWV0YWxrOHMuaW52YWxpZCIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJuYW1lIjoiYWRtaW4ifQ.wWHO5EbHhAm0Zc8SWaCiyIIBo5NlD6unxnnjafcYAIIx15Rudr79rjTfKjWsYQQIG_VrqdnFsxjaiL3r96aYkwint8kYOqtsF__rg0jybq3zPXTC28axWh3zkAhA3jJIh8FcMvo5YwjMCKm7i0GMPS3_nmIXTQnr3E_7zjH-prRG7U6-KJsM_Ccmo24dnbBEmpESXclK3JNJEK2jXDuxh_zWRrsESJwe89kwSjknRUj2qxVeUXWKi4nJVSq0x4oMut7O8enArMr2TQOzwOBnMH5xK-hAEHfL3DY7oWhbt_POTE463YV04vTeMxytf2gWCgmnkvF2w2qooIQ9hlrqDQ',
-//   access_token:
```
Comment (ChengYanJin) [inline ui/cypress/support/commands.js]: Maybe we should uncomment these in the previous commits `ui/cypress: Emulate an already logged in user`?

File: ui/cypress/support/commands.js [hunk 2]
```diff
@@ -51,18 +51,18 @@ Cypress.Commands.add('setupMocks', () => {
...
-//   token_type: 'bearer',
-//   scope:
-//     'openid profile email groups offline_access audience:server:client_id:oidc-auth-client',
-//   profile: {
-//     sub: 'CiQwOGE4Njg0Yi1kYjg4LTRiNzMtOTBhOS0zY2QxNjYxZjU0NjYSBWxvY2Fs',
-//     azp: 'metalk8s-ui',
-//     email: 'admin@metalk8s.invalid',
-//     email_verified: true,
-//     name: 'admin',
-//   },
```
Comment (ChengYanJin) [inline ui/cypress/support/commands.js]: Maybe we should uncomment these in the previous commits `ui/cypress: Emulate an already logged in user`?

File: ui/cypress/support/commands.js [hunk 3]
```diff
@@ -73,7 +73,7 @@ Cypress.Commands.add('login', () => {
         'oidc.user:/oidc:metalk8s-ui',
         JSON.stringify({
           ...ADMIN_JWT,
-        //   expires_at: Date.now() + 30000,
```
Comment (ChengYanJin) [inline ui/cypress/support/commands.js]: Same here?

**Review by ChengYanJin — COMMENTED** [review]
Great works! Thank you! 

Only a small comment ^^

File: ui/cypress/support/commands.js (hunk 2 above)
Comment (gdemonet) [reply ui/cypress/support/commands.js]: Oops you're right, bad fixup :)

**Review by ChengYanJin — DISMISSED** [review]
LGTM, thanks for the work!

**Review by ChengYanJin — APPROVED** [review]
LGTM! 
Appreciate all the work!

**Pull request merged**